extensions/in-memory/assetindex-memory/src/test/java/org/eclipse/dataspaceconnector/metadata/memory/InMemoryDataAddressResolverTest.java

```diff
@@ -39,6 +40,17 @@ void resolveForAsset_assetNull_raisesException() {
         assertThatThrownBy(() -> resolver.resolveForAsset(null)).isInstanceOf(NullPointerException.class);
     }
 
+    @Test
+    void resolveForAsset_whenAssetDeleted_raisesException() {
+        var testAsset = createAsset("foobar", UUID.randomUUID().toString());
+        var address = createDataAddress(testAsset);
+        resolver.accept(testAsset, address);
+        resolver.deleteById(testAsset.getId());
```

Comment:
```suggestion

        resolver.deleteById(testAsset.getId());
```
nit: a new line to separate test set up from tested method